scarlet/initialisation.py

```diff
@@ -0,0 +1,136 @@
+from .source import *
+from .interpolation import interpolate_observation
+from .observation import Observation
+from .wavelet import Starlet
+
+def build_initialisation_coadd(observations, wavelet = False):
+    """Build a channel weighted coadd to use for source detection
+
+    Parameters
+    ----------
+    observations: `~scarlet.observation.Observation`
+        Observation to use for the coadd.
+
+    Returns
+    -------
+    detect: array
+        2D image created by weighting all of the channels by SED
+    bg_cutoff: float
+        The minimum value in `detect` to include in detection.
+    """
+    try:
+        iter(observations)
+    except TypeError:
+        observations = [observations]
+    # The observation that lives in the same plane as the frame
+    loc = np.where([type(obs) is Observation for obs in observations])
+    # If more than one element is an `Observation`, then pick the first one as a reference (arbitrary)
+    obs_ref = observations[loc[0][0]]
+
+    n_images = 0
+    coadd = 0
+    jacobian = 0
+    weights = 0
+    for obs in observations:
+        try:
+            weights = np.array([w[w > 0].mean() for w in obs.weights])
+        except:
+            raise AttributeError(
+                "Observation.weights missing! Please set inverse variance weights"
+            )
+
+        if obs is obs_ref:
+            if wavelet is True:
+                images = Starlet(obs.images).filter()
+            else:
+                images = obs.images
+        else:
+            #interpolate low-res to reference resolution
+            images = interpolate_observation(obs, obs_ref.frame, wave_filter = wavelet)
+        n_images += len(images)
+        if wavelet is True:
+            coadd += np.sum(images / np.sum(images, axis=(-2,-1))[:, None, None], axis = 0)
+        else:
+            # Weighted coadd
+            coadd += (images * weights[:, None, None]).sum(axis = (0))
+            jacobian += weights.sum()
+
+    if wavelet is True:
+        coadd /= np.max(coadd)
+        bg_cutoff = 0
+        return coadd, bg_cutoff
+    coadd /= jacobian
+    # thresh is multiple above the rms of detect (weighted variance across channels)
+    bg_cutoff = np.sqrt((weights ** 2).sum()) / jacobian
+    return coadd, bg_cutoff
+
+def initialise(
```

Comment:
That function should not exist, and I'll show where the concept breaks.

Reply:
I agree. For reasons stated in #171 I think this goes against the current design pattern of scarlet. But something similar (and more general) would be useful to include in a separate repo that we make public to give users access to more specialized scripts.

Reply:
removed